lightning-background-processor/src/lib.rs

```diff
@@ -188,7 +189,7 @@ impl BackgroundProcessor {
 		PM: 'static + Deref<Target = PeerManager<Descriptor, CMH, RMH, L, UMH>> + Send + Sync,
 	>(
 		persister: PS, event_handler: EH, chain_monitor: M, channel_manager: CM,
-		net_graph_msg_handler: Option<NG>, peer_manager: PM, logger: L
+		net_graph_msg_handler: Option<NG>, peer_manager: PM, await_graph_sync_completion: bool, logger: L
```

Comment:
Need method-level docs that explains what the graph sync bool means - something about not pruning.

Reply:
`GraphSync` wouldn't necessarily need a reference to `NetworkGraph` -- that can still be taken as a parameter to `read_network_graph`. It would mainly be used to encapsulate the bool such that `read_network_graph` sets it as required instead of relying on the user to do it. Also, `BackgroundProcessor` just needs to read it, so we shouldn't allow for `BackgroundProcessor` to modify it by passing a bool.

As for the dependency, we can avoid it by having `BackgroundProcessor` take a closure instead of either a `GraphSync` reference as mentioned here or an `AtomicBool` as is currently done. Then the user would pass in `GraphSync::is_processing`.

Reply:
yeah, I do think it's cleaner to take a dependency, however, I think the current approach is very similar to the chain_sync_complete() method, and the interface is quite straightforward.

Reply:
I disagree. The current approach requires the caller to explicitly signal to `BackgroundProcessor` that the sync is complete whereas my alternative doesn't place that burden on the user.